Care-CureHospital/WebAppPatient/Validation/MedicalRecordValidation.cs

```diff
@@ -0,0 +1,253 @@
+                }
+                else if (!match.Success)
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool ValidateHealthInsuranceCard(string healthInsuranceCard)
+        {
+            try
+            {
+                Match match = healthInsuranceCardRegex.Match(healthInsuranceCard);
+
+                if (string.IsNullOrEmpty(healthInsuranceCard))
+                {
+                    return false;
+                }
+                else if (!match.Success)
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool ValidateDateOfBirth(DateTime dateOfBirth)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(dateOfBirth.ToString()))
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool ValidateContactNumber(string contactNumber)
+        {
+            try
+            {
+                Match match = contactNumberRegex.Match(contactNumber);
+
+                if (string.IsNullOrEmpty(contactNumber))
+                {
+                    return false;
+                }
+                else if (!match.Success)
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool ValidateEMail(string eMail)
+        {
+            try
+            {
+                Match match = eMailRegex.Match(eMail);
+
+                if (string.IsNullOrEmpty(eMail))
+                {
+                    return false;
+                }
+                else if (!match.Success)
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool EmptyAddress(MedicalRecordDto medicalRecordDto)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(medicalRecordDto.Patient.City.Name) || string.IsNullOrEmpty(medicalRecordDto.Patient.City.PostCode.ToString()) ||
+                    string.IsNullOrEmpty(medicalRecordDto.Patient.City.Address) || string.IsNullOrEmpty(medicalRecordDto.Patient.City.Country.Name))
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+    }
+}
+
```

Comment:
All of these validations should be extracted as separate classes, because it would prevent you from repeating yourself. Instead of repeating yourself you can just call the code you wrote before